android/Toodle/app/src/main/java/com/mozilla/toodle/rust/JNA.java

```diff
@@ -30,4 +30,53 @@
     long store_entid_for_attribute(Pointer Store, String attr);
 
     NativeResult store_sync(Pointer toodle, String userUuid, String serverUri);
+
+    // query
+    Pointer col_at_index(Pointer rows, int index);
+    Pointer col_at_index_as_long(Pointer rows, int index);
+    Pointer col_at_index_as_entid(Pointer rows, int index);
+    Pointer col_at_index_as_kw(Pointer rows, int index);
+    Pointer col_at_index_as_string(Pointer rows, int index);
+    Pointer col_at_index_as_uuid(Pointer rows, int index);
+    Pointer col_at_index_as_boolean(Pointer rows, int index);
+    Pointer col_at_index_as_double(Pointer rows, int index);
+    Pointer col_at_index_as_timestamp(Pointer rows, int index);
+
+    Pointer typed_value_as_int(Pointer value);
+    Pointer typed_value_as_long(Pointer value);
+    Pointer typed_value_as_entid(Pointer value);
+    Pointer typed_value_as_kw(Pointer value);
+    Pointer typed_value_as_string(Pointer value);
+    Pointer typed_value_as_uuid(Pointer value);
+    Pointer typed_value_as_boolean(Pointer value);
+    Pointer typed_value_as_double(Pointer value);
+    Pointer typed_value_as_timestamp(Pointer value);
+
+    Pointer row_at_index(Pointer rows, int index);
+    Pointer rows_iter(Pointer rows);
+    Pointer rows_iter_next(Pointer iter);
+
+    Pointer column_iter(Pointer rows);
```

Comment:
I'm not sure about the word 'column'. 'Coll' results consist of items or values. A 'rel' row contains bindings or values.